### DISCUSSION TITLE
Can't set title with "next/head" in mdxjs

### DISCUSSION THREAD

**Discussion by StarpTech** [vercel/next.js#33203] [Ideas]
# Bug report

## Describe the bug

As titled

## To Reproduce

```mdx
import Head from 'next/head'

<Head>
  <title>Privacy</title>
</Head>

# Privacy Policy
```

1. Checkout example https://github.com/vercel/next.js/tree/canary/examples/with-mdx
2. Use the `Head` as above.

## Expected behavior

Page title should be `Privacy`

## System information

```
"@mdx-js/loader": "^1.6.6",
"@next/mdx": "^9.4.4",
"next": "v9.4.5-canary.29",
```

**Comment by Tudes** [comment]
Hi @StarpTech .

Based on the description and requirements above you should be able to achieve that by removing the `<Head>` wrapper.

Adjusting `with-mdx` example, this will set a title for you page

```import Button from "../components/button.js";

<title>Privacy</title>

# MDX + Next.js

Look, a button! 👇

<Button>👋 Hello</Button>
```

**Comment by StarpTech** [comment]
Hi @Tudes thanks for the hint I didn't test it yet but `next/head` should be fully supported otherwise the nextjs plugin makes no sense. As a workaround, I use https://github.com/hashicorp/next-mdx-enhanced

**Comment by michrome** [comment]
You could approach this with the help of a component.

```js
// components/layout.js

import Head from "next/head";

export default function Layout({ title, children }) {
  return (
    <>
      <Head>
        <title>{title}</title>
      </Head>
      {children}
    </>
  );
}
```

```markdown
<!-- pages/privacy.mdx -->

import Layout from "../components/layout";

export const title = "Privacy";

<Layout title={title}>

# Privacy Policy

</Layout>
```

**Reply by MarcGruener** [reply]
I have been trying out the suggested solutions in this thread with the new Next.js App Router.
The solution by @Tudes seems to be working, however I can't seem to get yours working @michrome.

**HOWEVER:** It seems like the best and easiest solution for App Router is to leverage [Next.js Metadata](https://nextjs.org/docs/app/building-your-application/optimizing/metadata) like this:

```
<!-- blog/privacy/page.mdx -->

export const metadata = {
  title: 'Privacy',
  openGraph: {
    title: 'Privacy',
  },
}

# Privacy Policy

...
````

**Comment by rob-myers** [comment]
Seems the standard solution follows from the [NextJS MDX docs](https://nextjs.org/docs/pages/building-your-application/configuring/mdx#layouts):

```tsx
// foo.mdx

# Welcome

To my MDX page...

export { default } from '@/src/components/root';

// src/components/root.tsx
import Head from 'next/head'

export default function Root({ children }: { children: React.ReactNode }) {
  return (
    <>
      <Head><title>Foo bar baz</title></Head>
      {children}   
   </>
  );
}
```
